src/webapp/reports/autogenerated-forms/GridWithCatOptionCombos.tsx

```diff
@@ -0,0 +1,126 @@
+ *                     | Written Policy      |          |           |            |             |
+ *    ITNs - Extended  | Policy Implemented  |          |           |            |             |
+ *                     | Policy Extra        |          |           |            |             |
+ **/
+
+const GridWithCatOptionCombos: React.FC<GridWithCatOptionCombosProps> = props => {
+    const { dataFormInfo } = props;
+    const classes = useStyles();
+    const grid = React.useMemo(() => GridWithCatOptionCombosViewModel.get(props.section), [props.section]);
+
+    return (
+        <DataTableSection section={grid} dataFormInfo={dataFormInfo}>
+            <DataTable className={classes.table} layout="fixed" width="initial">
+                <TableHead>
+                    <DataTableRow>
+                        <DataTableColumnHeader width="400px" colSpan="2"></DataTableColumnHeader>
+
+                        {grid.columns.map(column => (
+                            <DataTableColumnHeader key={column.name}>
+                                <span>{column.name}</span>
+                            </DataTableColumnHeader>
+                        ))}
+                    </DataTableRow>
+                </TableHead>
+
+                <TableBody>
+                    {grid.rows.map(row => {
+                        const groupName = row.groupName;
+                        const rows = row.rows;
+
+                        return rows.map((row, idx) => (
+                            <DataTableRow key={`${groupName}-${idx}`}>
+                                {groupName ? (
+                                    <>
+                                        {idx === 0 && (
+                                            <DataTableCell
+                                                className={classes.rowTitle}
+                                                rowSpan={rows.length.toString()}
+                                            >
+                                                <span>{groupName}</span>
+                                            </DataTableCell>
+                                        )}
+
+                                        <DataTableCell>
+                                            <span>{row.deName}</span>
+                                        </DataTableCell>
+                                    </>
+                                ) : (
+                                    <DataTableCell colSpan="2">
+                                        <span>{row.deName}</span>
+                                    </DataTableCell>
+                                )}
+
+                                {grid.columns.map(column => {
+                                    const dataElement = column.items.find(item =>
+                                        item.row.rows.map(row => row.deName).includes(row.deName)
+                                    )?.dataElement;
```

Comment:
It's a good practice to not include logic inside UI component. For this case I think we can move the code to the `GridWithCatOptionCombosViewModel.ts` file

Also here we're having the same dataElement for all the **Microscopy (All ages, both active and passive)** section. You can inspect the ids in devtools:

This is correct the dataElement id for **Microscopy (All ages, both active and passive) - Tested** is **DPNsg3wjxGm**
![image](https://github.com/EyeSeeTea/d2-autogen-forms/assets/461124/d3403884-7a20-4e33-884a-ea92ce2d6b56)

but if you look the other rows, all of them (except **Mixed (more than one species detected)**) are being assigned to the same data element.

![image](https://github.com/EyeSeeTea/d2-autogen-forms/assets/461124/87f45e15-c6fe-42ac-ad8e-f892f9563e90)

For dataElement **Microscopy (All ages, both active and passive) - Positive** it should be the ID **enVf9I39FRZ**